.github/workflows/receive-pr.yml

```diff
@@ -0,0 +1,46 @@
+# -------------------------------------------------------------------------------------
+#
+# Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+#
+# WSO2 LLC. licenses this file to you under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# --------------------------------------------------------------------------------------
+
+# This workflow will receive a PR and save the PR number for later use.
+
+name: Receive PR
+
+on:
+    pull_request:
+        branches: [master]
+
+jobs:
+    save-pr-information:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout Repository
+              uses: actions/checkout@v3
+
+            - name: Display PR Information
+              run: echo "PR Number \#${{github.event.number}}"
+
+            - name: Save PR Number for Later Use
+              run: echo "${{github.event.number}}" > PR_NUMBER
+
+            - name: Upload PR Number as Artifact
+              uses: actions/upload-artifact@v3
```

Comment:
Let's use the v4, as according to Github, it is said to be more performant than v3[1].

[1] https://github.blog/2024-02-12-get-started-with-v4-of-github-actions-artifacts/#whats-improved-in-v4

Reply:
Since this is a tiny file upload, I don't think version upgrade will impact much.